From 463ba656d159fa7e8b8c89cf253ee77198224eec Mon Sep 17 00:00:00 2001
From: Sanket Goswami <Sanket.Goswami@amd.com>
Date: Thu, 28 Oct 2021 17:09:34 +0530
Subject: [PATCH 47/58] SUNLIGHT: platform/x86: amd-pmc: Store the pci_dev
 instance inside struct amd_pmc_dev

Description:
 - Store the root port information in amd_pmc_probe() so that the
   information can be used across multiple routines.

Bug: N/A
Change-Id: Ic8f654d80adf99d04634771d7b5006eef0a9d9be
Signed-off-by: Ionut Nechita <ionut_n2001@yahoo.com>
Signed-off-by: Sanket Goswami <Sanket.Goswami@amd.com>
---
 drivers/platform/x86/amd-pmc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/platform/x86/amd-pmc.c b/drivers/platform/x86/amd-pmc.c
index 4d8aa22a909f..25f6aa6f1e86 100644
--- a/drivers/platform/x86/amd-pmc.c
+++ b/drivers/platform/x86/amd-pmc.c
@@ -121,6 +121,7 @@ struct amd_pmc_dev {
 	u16 minor;
 	u16 rev;
 	struct device *dev;
+	struct pci_dev *rdev;
 	struct mutex lock; /* generic mutex lock */
 #if IS_ENABLED(CONFIG_DEBUG_FS)
 	struct dentry *dbgfs_dir;
@@ -539,6 +540,7 @@ static int amd_pmc_probe(struct platform_device *pdev)
 	}
 
 	dev->cpu_id = rdev->device;
+	dev->rdev = rdev;
 	err = pci_write_config_dword(rdev, AMD_PMC_SMU_INDEX_ADDRESS, AMD_PMC_BASE_ADDR_LO);
 	if (err) {
 		dev_err(dev->dev, "error writing to 0x%x\n", AMD_PMC_SMU_INDEX_ADDRESS);
-- 
2.35.1

